src/providers/tpm_provider/utils.rs

```diff
@@ -99,11 +100,18 @@ pub fn parsec_to_tpm_params(attributes: Attributes) -> Result<KeyParams> {
                 x @ 1024 | x @ 2048 | x @ 3072 | x @ 4096 => x.try_into().unwrap(), // will not fail on the matched values
                 _ => return Err(ResponseStatus::PsaErrorInvalidArgument),
             };
-            Ok(KeyParams::Rsa {
-                size,
-                scheme: convert_asym_scheme_to_tpm(attributes.policy.permitted_algorithms)?,
-                pub_exponent: 0,
-            })
+            if attributes.is_encrypt_permitted() {
+                Ok(KeyParams::RsaEncrypt {
+                    size,
+                    pub_exponent: 0,
+                })
+            } else {
+                Ok(KeyParams::RsaSign {
+                    size,
+                    scheme: convert_asym_scheme_to_tpm(attributes.policy.permitted_algorithms)?,
+                    pub_exponent: 0,
+                })
+            }
```

Comment:
I'm wondering if this should be an `else if attributes.is_hash_signable() || attributes.is_hash_verifiable()` with the fallback case something like `Err(ResponseStatus::PsaErrorNotSupported)`.